R/dependencies.R

```diff
@@ -716,6 +717,30 @@ renv_dependencies_discover_r_pacman <- function(node, envir) {
 
 }
 
+renv_dependencies_discover_r_import <- function(node, envir) {
+
+  node <- renv_call_expect(node, "import", c("from", "here", "into"))
+  if (is.null(node))
+    return(FALSE)
+
+  # match a call to any function with a '.from' argument
+  prototype <- function(.from, ...) {}
```

Comment:
I think you'll need to use a different prototype depending on which function is being used. E.g.

```
> args(import::from)
function (.from, ..., .into = "imports", .library = .libPaths()[1L],  .character_only = FALSE) 
> args(import::into)
function (.into, ..., .from, .library = .libPaths()[1L], .character_only = FALSE) 
> args(import::here)
function (..., .from, .library = .libPaths()[1L], .character_only = FALSE) 
```

In other words, unless we're using `import::from`, I think the prototype has to be:

```
prototype <- function(..., .from)
```

Reply:
I don't seem to be able to construct a valid call to import::from or import::here that would fail to match the given prototype. For example, any of the following match:
```
proto = function(.from, ...) {}

match.call(proto, expression(import::from(package, func)))
match.call(proto, expression(import::from(func, .from = package)))

match.call(proto, expression(import::here(func1, func2, .from = package)))
match.call(proto, expression(import::into(func1, func2, .into = "imports:pkg", .from = package)))
match.call(proto, expression(import::into("imports:pkg", func1, func2, .from = package)))
```
while constructions like the following that would otherwise fail to match aren't valid import calls (``Argument `.from` must be specified``)
```
import::here(func, package)
import::here(func1, func2, package)
import::into(func1, func2, package, .into = "imports:pkg")
```
So I think the prototype should cover all uses of this import mechanism, unless I'm missing something.

Reply:
My main worry is that this code would cause `renv` to falsely infer a dependency:

```
> proto <- function(.from, ...) {}
> match.call(proto, quote(import::here("object")))
import::here(.from = "object")
```

This is of course an invalid use of `import::here()` by the user, but I think it's worth guarding against.

Reply:
In other words: I think it's worth being resilient against certain inappropriate usages of these functions.